src/CommunityToolkit.Maui/Views/Popup/PopupPage.shared.cs

```diff
@@ -25,13 +30,18 @@ partial class PopupPage : ContentPage
 	readonly Command tapOutsideOfPopupCommand;
 	readonly WeakEventManager popupClosedEventManager = new();
 
-	public PopupPage(View view, IPopupOptions popupOptions)
-		: this(view as Popup ?? CreatePopupFromView<Popup>(view), popupOptions)
+	public PopupPage(View view, IPopupOptions popupOptions, IDictionary<string,object>? shellParameters)
+		: this(view as Popup ?? CreatePopupFromView<Popup>(view), popupOptions, shellParameters)
 	{
 		ArgumentNullException.ThrowIfNull(view);
+
+		if (view is IQueryAttributable queryAttributableView && shellParameters is not null)
```

Comment:
Why do we need to apply this is we already pass the parameters into `Shell.Current.GoToAsync()`?

Reply:
Good catch! Long story short, I dug into the MAUI source code and found that the Shell navigation engine only fires `IQueryAttributable.ApplyQueryAttributes()` when `IQueryAttributable` is inherited from the `ContentPage` or the `ContentPage.BindingContext` (View Model). 

This is great for us because the Shell Navigation engine will automatically fire `IQueryAttributable.ApplyQueryAttributes()` on the Popup View Model because we propogate its `BindingContext` to `PopupPage.BindingContext`.

However, because the Popup/View given to us from the user is not used as the underlying `ContentPage` (we use our internal `PopupPage` as the underlying `ContentPage`), we have to manually check for `IQueryAttributable` and call `IQueryAttributable.ApplyQueryAttributes()`.

Reply:
All that said - this conversation inspired me to just come up with a new mechanism for manually calling `IQueryAttributable.ApplyQueryAttributes()` **from** IQueryAttributable.ApplyQueryAttributes()! 

Instead of doing it in the constructor here, and requiring us to pass in `IDictionary<string,object>` into the Constructor of `PopupPage`, we can instead let the Shell Navigation engine provide the `IDictionary<string,object>` to us by having `PopupPage` inherit from `IQueryAttributable` (`PopupPage :IQueryAttributable`).

Now, when `PopupPage.ApplyQueryAttributes(IDictionary<string, object>)` fires we can check if the `Popup` or `Popup.Content` is inheriting from `IQueryAttributable` and pass in the `IDictionary<string, object>` provided to us from the Shell Navigation engine.

Reply:
Thanks for explaining. Yes I like the idea of implementing the interface and doing it your proposed way